### DISCUSSION TITLE
text shadow & border

### DISCUSSION THREAD

**Discussion by sreejitaduttaa** [fastn-stack/fastn#1333] [Ideas & RFCs]
can we have commands to style text , as in add border and shadow?

**Comment by amitu** [comment]
@harshdoesdev maybe you can take a stab at this one. Should be easy. Even better if you can write some instructions and @sreejitaduttaa follows them and sends the PR herself.

**Reply by harshdoesdev** [reply]
Sure @amitu!

Hey, @sreejitaduttaa! I've created a pull request (PR) for implementing this feature. We're not merging it just yet because we'd like you to take the lead. After reviewing my PR, please create your own PR to implement this feature. I've outlined the steps you should follow to set up the development environment correctly and provided links to some guides related to Rust and Git. Happy coding!

The steps:

1. Install rust and other dev tools (git, etc):

Here is the rust's official installation guide: https://www.rust-lang.org/tools/install
and this is the official guide for installing git: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git

3. Setup VS Code: you can install the `rust-analyzer` plugin for VS Code for better language support

4. Make a fork of `fastn` and clone it on your local system

5. Observe this PR and try to replicate: https://github.com/fastn-stack/fastn/pull/1334

- In the `fastn-js/src/property.rs` we define `text-shadow` as a new property kind
- then in `ftd/src/interpreter/things/default.rs` we define `text-shadow` as an argument for text component (and also other text-like components), we are making it a record kind of type ftd::interpreter::FTD_SHADOW, and making it optional
- then in the `ftd/src/js/element.rs` we write some code for including the `text-shadow` in the component statements of text-like elements.
- then in the `fastn-js/js/dom.js` we write some code which will be executed when the fastn code has been converted into Javascript code. this code is executed during the SSR phase and Hydration phase (basically when the webpage is `rendered`). we defined TextShadow as a property kind in Javascript and also wrote a function which will attach the text-shadow when a property of type "TextShadow" is encountered.

Hope this was helpful and feel free to ask any questions in the discord channel. We would be very happy to help you and welcome a new contributor.

**Reply by sreejitaduttaa** [reply]
yeah sure, I'll do it tomorrow!
